source/particle/output/interface.cc

```diff
@@ -100,19 +100,12 @@ namespace aspect
           prm.enter_subsection ("Tracers");
           {
             name = prm.get ("Data output format");
-
-#ifndef DEAL_II_HAVE_HDF5
-            AssertThrow (name != "hdf5",
-                         ExcMessage ("deal.ii was not compiled with HDF5 support, "
```

Comment:
I moved it into the Constructor of hdf5.cc to keep the interface.cc clean of specific plugin names. It will still throw in the same function and with the same error if somebody tries to use hdf5 without having deal.II compiled with it.